Up2dateService/Up2dateService/SetupManager/SetupManager.cs

```diff
@@ -264,15 +378,51 @@ private void RefreshPackageList()
                     updatedPackage.InstallDate = null;
                     updatedPackage.EstimatedSize = null;
                     updatedPackage.UrlInfoAbout = null;
-                    if (updatedPackage.Status != PackageStatus.Downloading && updatedPackage.Status != PackageStatus.Installing)
-                    {
+                    if (updatedPackage.Status != PackageStatus.Downloading &&
+                        updatedPackage.Status != PackageStatus.Installing)
                         updatedPackage.Status = PackageStatus.Downloaded;
-                    }
                 }
+
+                if (updatedPackage.Filepath.Contains(NugetExtension))
+                    using (var zipFile = ZipFile.OpenRead(updatedPackage.Filepath))
+                    {
+                        var nuspec = zipFile.Entries.First(zipArchiveEntry => zipArchiveEntry.Name.Contains(".nuspec"));
+                        using (var nuspecStream = nuspec.Open())
+                        {
+                            using (var sr = new StreamReader(nuspecStream, Encoding.UTF8))
+                            {
+                                var xmlData = sr.ReadToEnd();
+                                var doc = new XmlDocument();
+                                doc.LoadXml(xmlData);
+                                updatedPackage.DisplayName = doc.GetElementsByTagName("title")[0].InnerText;
+                                updatedPackage.ProductName = doc.GetElementsByTagName("id")[0].InnerText;
+                                updatedPackage.DisplayVersion = doc.GetElementsByTagName("version")[0].InnerText;
+                                updatedPackage.Publisher = doc.GetElementsByTagName("authors")[0].InnerText;
+                            }
+                        }
+
+                        using (var ps = PowerShell.Create())
+                        {
+                            const string psCommand = @"choco list -li";
+                            ps.AddScript(psCommand);
+                            var result = ps.Invoke<string>();
+                            if (result.Any(item => item.Contains(updatedPackage.ProductName)))
+                                updatedPackage.Status = PackageStatus.Installed;
+                        }
+                    }
+
                 lockedPackages[i] = updatedPackage;
             }
 
             SafeUpdatePackages(lockedPackages);
         }
+
+        private static bool IsChocoInstalled()
+        {
+            var ps = PowerShell.Create();
+            ps.AddScript("choco --version");
+            var value = ps.Invoke<string>();
+            return value.Count > 0;
```

Comment:
I guess, if choco is absent then PS will return an error message, so value.Count would not be zero anyway?

Reply:
On Powershell.AddScript the error messages are going into CERR pipeline and when error occures there is no text at all.
On the other hand when use PowerShell.AddCommand errors are translated to exception